pkg/ctrl/manager.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctrl
+
+// DefaultControllerManager is the default ControllerManager used for package functions
+var DefaultControllerManager = &ControlleManager{}
+
+// ControllerManager manages running Controllers
+type ControlleManager struct {
+	// Stop may be closed to shutdown all Controllers
+	// Defaults to a new channel if unspecified
+	Stop <-chan struct{}
+
+	controllers []*Controller
+}
+
+// RegisterController registers a controller to be started with Start.
+// The Stop channel will be set on any Controllers where it is unspecified.
+func (cm ControlleManager) RegisterController(c *Controller) {
+	cm.controllers = append(cm.controllers, c)
+}
+
+// Start starts all the controllers
```

Comment:
As discussed before, document that this is blocking.

Reply:
Done